website/docs/concepts/provider_lifecycles.mdx

```diff
@@ -0,0 +1,102 @@
+- Alive
+- Paused
+- Disposed
+
+### Disposed / Uninitialized
+
+An **Uninitialized** or **Disposed** provider does not take up any memory since its state is not initialized.
+Essentially it is just a definition of how to create the provider's state when you need it.
+It will stay that way until an **Alive** provider or a [WidgetRef] from the UI reads, watches, or listens to it.
+
+### Creating -> Alive
+
+When an **Uninitialized** provider is read, listened to or watched it's state will be created.
+
+During creation your provider's build function will be run. 
+Any providers that you read or watch using the `ref` exposed by the callback will be created as needed and their state will be retrieved.
+
+If there are any circular dependencies during this creation process Riverpod will throw an error.
+The best way to fix this error is to redesign your dependencies to have a uni-directional dataflow.
+
+The provider's state is stored in a [ProviderContainer]. In a Flutter app this container is in a [ProviderScope] widget.
+
+As such, even though the definition of how to create the state (the provider) is global, the state is actually local,
+and can be different in different portions of your UI using nested [ProviderScope] widgets and overrides.
+
+This is very similar to how flutter widgets work. You only pay for the definition once, but can reuse the state in different parts of the tree as needed.
+
+### Alive
+
+When your provider is **Alive**, changes to its state will cause dependent providers and/or the dependent UI to rebuild.
+
+From the other perspective, as a reactive framework, you can watch other providers to have the provider recreate itself whenever one of it's dependencies changes.
+
+If you need to have some long-lived state that depends on other state you can use Ref's [listen] method to subscribe for changes on another provider without causing a rebuild of the provider.
+
+If you need to use the state from another provider in a side-effect, you can use Ref's [read] method to obtain the current state from another provider.
+
+Typically when constructing a [StateNotifier] or [ChangeNotifier] class you should pass in the `ref` to allow the Notifier to obtain the current value of dependencies as needed. By using the new [Notifier] and [AsyncNotifier] classes from Riverpod 2.0, the ref is already available as an instance member of the class.
+
+### Alive -> Paused
+When an **Alive** provider is no longer listened to by other providers or the UI, it goes into a **Paused** state. 
+This means that it no longer will react to changes on providers it is listening to.
+This is an optimization, as if you are not listening to the provider, there is no need to keep it alive. 
+Every provider not being used will be returned to a **Paused** state, reducing the computational burden of your app.
+
+If you need to keep a provider alive for side-effects, make sure to listen to it in an appropriate place in the UI where it should be kept alive.
+
```

Comment:
What is meant by side-effects here? In other words, what kinds of computations / actions can I still expect to be executed (at unpausing), which ones not?

Unrelated to this PR, but maybe it would be nice to have an imperative way to keep providers alive (listening to them purely for side-effects seems a bit dirty)

Reply:
Side-effects meaning the provider does more than just hold a value.

It will not cancel / pause timers / network requests if you create those in your providers (unless you cancel them yourself in the onCancel/onDispose ref hook).

I agree that currently this part of the API feels a bit dirty, but side-effects are kind of dirty anyways, this just encourages you to think carefully about side-effects so that you don't waste computation time unless you are in a portion of the UI that needs it.

I typically create a helper method for this purpose, and then call keepAlive at the top of the widget tree, or the portion of the UI that I need it to be alive for:
```dart
extension KeepAliveRef on WidgetRef {
  void keepAlive<T>(ProviderListenable<T> provider) => listen<T>(provider, (T? _, T __) {});
}
```

Reply:
Thanks for the detailed explanation - and I already needed this snippet! (Update the GPS location when location access granted.)

... maybe an API for this wouldn't be too bad @rrousselGit